### PR TITLE
Analytics: Adds IntersectionObserver functionality to IframeTransport

### PR DESCRIPTION
**Pull request by jonkeller**
Paired with #11723

### CODE REVIEW

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -178,6 +179,12 @@ export class AmpAnalytics extends AMP.BaseElement {
 
   /** @override */
   unlayoutCallback() {
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Ignore the following 6 lines. They're just a copy of #11713, and will be rebased away when that one merges.

File: extensions/amp-analytics/0.1/iframe-transport-intersection-observer.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            this.creativeIdToIntersectionObserver_[reqCreativeId].observe(
+                this.creativeIdToTarget_[reqCreativeId]);
+          }
+        });
+  }
+
+  /**
+   * Adds a creativeId to the list of ids we are listening to
+   * IntersectionObserver events for.
+   * @param {!string} creativeId The creativeId used to identify <the
```
Comment (glevitzky) [inline extensions/amp-analytics/0.1/iframe-transport-intersection-observer.js]: Nit: No need for the `!` on primitive types.

File: extensions/amp-analytics/0.1/iframe-transport.js [hunk 3]
```diff
@@ -116,12 +130,14 @@ export class IframeTransport {
...
-    const frame = createElementWithAttributes(this.ampWin_.document, 'iframe',
-        /** @type {!JsonObject} */ ({
-          sandbox: 'allow-scripts allow-same-origin',
-          name: frameName,
-          'data-amp-3p-sentinel': sentinel,
-        }));
+
+    const frame = /** @type {!HTMLIFrameElement} */
+        (createElementWithAttributes(this.ampWin_.document, 'iframe',
+            /** @type {!JsonObject} */ ({
```
Comment (glevitzky) [inline extensions/amp-analytics/0.1/iframe-transport.js]: You can avoid the cast here by using dict().

File: extensions/amp-analytics/0.1/iframe-transport-intersection-observer.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
+/**
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/iframe-transport-intersection-observer.js]: TODO: Unit tests

File: extensions/amp-analytics/0.1/iframe-transport.js (hunk 3 above)
Comment (jonkeller) [reply extensions/amp-analytics/0.1/iframe-transport.js]: Done

File: extensions/amp-analytics/0.1/iframe-transport-intersection-observer.js (hunk 2 above)
Comment (jonkeller) [reply extensions/amp-analytics/0.1/iframe-transport-intersection-observer.js]: Done

**Comment by jonkeller** [comment]
This is outdated. The work was performed by #12217 and #11724

**Pull request closed without merging**